src/bech32.h

```diff
@@ -21,6 +21,9 @@
 namespace bech32
 {
 
+/** The Bech32 and Bech32m checksum size */
+constexpr size_t CHECKSUM_SIZE = 6;
```

Comment:
IIUC, I don't think static makes a difference here since we aren't talking about a class member? Regarding `CHARSET` and `CHARSET_REV`, what's the advantage of having them in the header file if we aren't using them outside of bech32.cpp?

Reply:
>  I don't think static makes a difference here

Based on other usages in the code, `static constexpr` in headers is good practice to prevent potential linking issues and ensure each translation unit has its own independent copy. Let me know if I'm wrong here.

> what's the advantage of having them in the header file

Separating constants and definitions into the header, leaving implementations in the cpp, i.e. separation of concerns.

If you think this is outside the scope, I can accept that, thanks for considering my suggestions.

Reply:
@sipa , @paplorinc thanks for the explanation! In this case, since we are in a _named_ namespace, it seems the static keyword would be preferable? Or doesn't matter? Will do some more reading on this cause I'm not sure what the best practice is here.

> Separating constants and definitions into the header, leaving implementations in the cpp, i.e. separation of concerns.

This is another one I'm not sure about: in my mind, `CHARSET` is internal to bech32 hence why it makes sense to keep it in bech32.cpp. But I suppose you could also argue the char set is a definition (from BIP173), so it should be in the header. Either way, I think it's out of scope for this PR, but thought it was an interesting point.

Reply:
Should I add this to https://github.com/bitcoin/bitcoin/pull/30623#issuecomment-2280136628 as well?